src/main/java/dev/morling/onebrc/CalculateAverage_iziamos.java

```diff
@@ -0,0 +1,345 @@
+
+        aggregate.forEach((name, max, min, sum, count) -> output.put(bytesToString(name), new ResultRow(min, (double) sum / count, max)));
+
+        System.out.println(output);
+//        System.out.println(Arrays.stream(aggregate.counts).sum());
+    }
+
+    private static void mergeResults(final PartialResult aggregate, final PartialResult result) {
+        mergeLock.lock();
+        aggregate.merge(result);
+        mergeLock.unlock();
+    }
+
+    private record ResultRow(long min, double mean, long max) {
+        public String toString() {
+            return formatLong(min) + "/" + round(mean) + "/" + formatLong(max);
+        }
+
+        private double formatLong(final long value) {
+
+            return value / 10.0;
+        }
+        private double round(double value) {
+            return Math.round(value) / 10.0;
+        }
+    }
+
+    private static CompletableFuture<PartialResult> processSegment(final FileChannel channel,
+                                                                   final long chunkNumber,
+                                                                   final long size,
+                                                                   final boolean isLast) {
+
+        final var result = new CompletableFuture<PartialResult>();
+
+        Thread.ofVirtual().start(() -> {
+            try {
+                final long start = chunkNumber * size;
+                final long memoryMapSize = mapsize(channel.size(), start, size, isLast);
+                final ByteBuffer mmap = channel.map(READ_ONLY, start, memoryMapSize);
+                skipIncomplete(mmap, start);
+                result.complete(processEvents(mmap, isLast ? memoryMapSize : size));
+            }
+            catch (IOException e) {
+                result.completeExceptionally(e);
+            }
+        });
+
+        return result;
+    }
+
+    private static long mapsize(final long total, final long start, final long chunk, final boolean isLast) {
+        final long chunkWithSomeOverlap = chunk + 128;
+        if (isLast) {
+            return total - start;
+        }
+
+        return chunkWithSomeOverlap;
+    }
+
+    private static void skipIncomplete(final ByteBuffer buffer, final long start) {
+        if (start == 0) {
+            return;
+        }
+        for (byte b = buffer.get();; b = buffer.get()) {
+            if (b == '\n')
+                return;
+        }
+    }
+
+    private static PartialResult processEvents(final ByteBuffer buffer, final long limit) {
+        final var result = new PartialResult();
+        while (buffer.hasRemaining() && buffer.position() <= limit) {
+            processEvent(buffer, result);
+        }
+
+        return result;
+    }
+
+    private static void processEvent(final ByteBuffer buffer, final PartialResult map) {
+        final long hash = parseId(buffer);
+        final int value = readValue(buffer);
+
+        map.recordResult(hash, value);
```

Comment:
How do you deal with hash collisions here?

Reply:
I deliberately ignore that. I'm guessing we need something in the rules maybe. I'll make changes